src/Runner.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -634,28 +634,33 @@ private async Task<GitHubRunnerRegisterToken> GetJITRunnerTokenAsync(string gith
                             var jsonResponse = await response.Content.ReadAsStringAsync();
                             return StringUtil.ConvertFromJson<GitHubRunnerRegisterToken>(jsonResponse);
                         }
-                        else if(response.StatusCode == System.Net.HttpStatusCode.NotFound)
-                        {
-                            // It doesn't make sense to retry in this case, so just stop
-                            break;
-                        }
                         else
                         {
                             _term.WriteError($"Http response code: {response.StatusCode} from 'POST {githubApiUrl}'");
                             var errorResponse = await response.Content.ReadAsStringAsync();
                             _term.WriteError(errorResponse);
-                            response.EnsureSuccessStatusCode();
+
+                            if(response.StatusCode == System.Net.HttpStatusCode.NotFound)
+                            {
+                                _term.WriteError("Invalid repository URL or register token");
+                                return StringUtil.ConvertFromJson<GitHubRunnerRegisterToken>(errorResponse);
```

Comment:
`ConvertFromJson` will return useless data if the `errorResponse` is not a contract of `GitHubRunnerRegisterToken`

Reply:
I changed this in the new commit.